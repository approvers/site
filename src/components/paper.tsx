import type { ReactNode } from "react";
import styles from "../scss/components/paper.module.scss";

const Paper = ({ children }: { children: ReactNode }): JSX.Element => (
  <div className={styles.paperWrapper}>
    <div className={styles.paper}>{children}</div>
  </div>
);

export const Grid = ({ children }: { children: readonly ReactNode[] }): JSX.Element => (
  <div className={styles.grid}>
    {children.map((child, index) => (
      <Paper key={index}>{child}</Paper>
    ))}
  </div>
);
