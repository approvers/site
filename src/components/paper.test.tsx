import { Grid } from "./paper";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Grid>
        <div>hoge</div>
        <div>fuga</div>
        <div>foo</div>
        <div>bar</div>
      </Grid>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
