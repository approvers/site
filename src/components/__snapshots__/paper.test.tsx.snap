// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="grid"
>
  <div
    className="paperWrapper"
  >
    <div
      className="paper"
    >
      <div>
        hoge
      </div>
    </div>
  </div>
  <div
    className="paperWrapper"
  >
    <div
      className="paper"
    >
      <div>
        fuga
      </div>
    </div>
  </div>
  <div
    className="paperWrapper"
  >
    <div
      className="paper"
    >
      <div>
        foo
      </div>
    </div>
  </div>
  <div
    className="paperWrapper"
  >
    <div
      className="paper"
    >
      <div>
        bar
      </div>
    </div>
  </div>
</div>
`;
